src/simd/simd_f64_return_nans.rs

```diff
@@ -0,0 +1,628 @@
+use super::config::SIMDInstructionSet;
+use super::generic::{SIMDArgMinMax, SIMDOps};
+#[cfg(target_arch = "aarch64")]
+use std::arch::aarch64::*;
+#[cfg(target_arch = "arm")]
+use std::arch::arm::*;
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use std::arch::x86_64::*;
+
+use super::task::{max_index_value, min_index_value};
+
+const XOR_VALUE: i64 = 0x7FFFFFFFFFFFFFFF; // i64::MAX
+const BIT_SHIFT: u32 = 63;
+
+#[inline(always)]
+fn _i64ord_to_f64(ord_i64: i64) -> f64 {
+    // TODO: more efficient transformation -> can be decreasing order as well
+    let v = ((ord_i64 >> BIT_SHIFT) & XOR_VALUE) ^ ord_i64;
+    unsafe { std::mem::transmute::<i64, f64>(v) }
+}
+
+const MAX_INDEX: usize = i64::MAX as usize;
+
+// ------------------------------------------ AVX2 ------------------------------------------
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+mod avx2_float_return_nan {
+    use super::super::config::{AVX2FloatReturnNaN, AVX2};
+    use super::*;
+
+    const LANE_SIZE: usize = AVX2::LANE_SIZE_64;
+    const XOR_MASK: __m256i = unsafe { std::mem::transmute([XOR_VALUE; LANE_SIZE]) };
+
+    #[inline(always)]
+    unsafe fn _f64_to_i64ord(f64_as_m256i: __m256i) -> __m256i {
+        // on a scalar: ((v >> 63) & 0x7FFFFFFFFFFFFFFF) ^ v
+        // TODO _mm256_srai_epi64 is not available on AVX
+        // https://stackoverflow.com/questions/28807341/simd-signed-with-unsigned-multiplication-for-64-bit-64-bit-to-128-bit
+        let sign_bit_shifted = _mm256_srai_epi64(f64_as_m256i, BIT_SHIFT);
```

Comment:
This is currently the largest open issue at the moment..

Reply:
As alternative we could look into https://doc.rust-lang.org/beta/core/arch/x86_64/fn._mm256_srli_epi64.html#

Reply:
I came up with this solution - leveraging the fact that we are only interested in the sign bit (i.e., right-shift of 63 bits while shifting in sign bits):
```rs
// perform a 32-bit right shift (while shifting in sign bits)
_mm256_srai_epi32(f64_as_m256i, BIT_SHIFT as i32);
// copy every even 32-bit part to the next odd 32-bit part
let sign_bit_masked = _mm256_and_si256(sign_bit_shifted, XOR_MASK);
```